test-cypress/integration/monitor/monitor.resources.spec.js

```diff
@@ -0,0 +1,55 @@
+describe('Monitor Resources', () => {
+
+    let repositoryId;
+
+    before(() => {
+        repositoryId = 'monitoring-repo' + Date.now();
+        cy.createRepository({id: repositoryId});
+    });
+
+    beforeEach(() => {
+        cy.presetRepositoryCookie(repositoryId);
+
+        cy.visit('/monitor/resources');
+        // Ensure the chart on the default active tab is rendered
+        getActiveTabContent().find('svg').should('be.visible');
+    });
+
+    after(() => {
+        cy.deleteRepository(repositoryId);
+    });
+
+    function getTabsPanel() {
+        return cy.get('.graphics');
+    }
+
+    function getActiveTab() {
+        return getTabsPanel().find('.nav-link.active');
+    }
+
+    function getTabContent() {
+        return cy.get('.tab-content');
+    }
+
+    function getActiveTabContent() {
+        return getTabContent().find('.tab-pane.active');
+    }
+
+    function getMemoryUsageChart() {
+        return getTabContent().find('.nv-stackedAreaChart');
+    }
+
+    it('Initial state ', () => {
+        // Graphics container should be present
+        getTabsPanel().should('be.visible');
+        // All tabs should be visible
+        let tabs = ['Memory', 'Threads', 'CPU', 'Classes'];
+        getTabsPanel().find('.nav-item').should('have.length', 4).each(($tab, index) => {
```

Comment:
An alternative
```
tabs.forEach((tabName, index) => {
  getTabsPanel().find('.nav-item').eq(index).should('be.visible').and('contain', tabName);
})
```
although it needs separate check for the length

Reply:
yeah, that and also this will make 4 selections in the DOM instead of 1